hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/RingbufferContainer.java

```diff
@@ -234,15 +242,22 @@ public long remainingCapacity() {
      *                                         deserialized
      */
     public long add(T item) {
-        final long sequence = addInternal(item);
+        final long nextSequence = ringbuffer.tailSequence() + 1;
```

Comment:
Would rather have this inside the `Ringbuffer` implementation. Something like `ringbuffer.peekNextTailSequence()` to adhere to common concepts, and keep this hidden as an implementation detail.

Reply:
Makes sense, changing accordingly.